qcodes/dataset/plotting.py

```diff
@@ -131,7 +135,7 @@ def plot_by_id(run_id: int,
             xpoints = data[0]['data'][order]
             ypoints = data[1]['data'][order]
```

Comment:
@WilliamHPNielsen noticed a bug - this sorting should be happening only for the 'line' plots; it definitely should not happen for the string-valued arrays.

Reply:
as discussed with @QCoDeS/core , the solution is to make the sorting only relevant for 'line' plot, since for 'scatter' it is not needed. (the other option was to 'not sort' if 'x' is string-valued.)